Prevent krunner from crashing by returning early before entering broken code.
The exact cause of the crash isn't known yet, although it seems Musl specific.

This is not great, but bad search results are not as bad as crashes.

https://bugs.kde.org/show_bug.cgi?id=511233

---
diff --git a/runners/services/bitap.h b/runners/services/bitap.h
index 37c1f34557..ca37e95300 100644
--- a/runners/services/bitap.h
+++ b/runners/services/bitap.h
@@ -42,6 +42,8 @@ inline std::optional<Match> bitap(const QStringView &name, const QStringView &pa
         return Match{.size = pattern.size(), .distance = 0}; // Perfect match
     }
 
+    return std::nullopt;
+
     if (pattern.isEmpty() || name.isEmpty()) {
         return std::nullopt;
     }
